### PR TITLE
Adding JSONSchema Validator and Default Schema

### PR DESCRIPTION
**Pull request by QasimAbbas**
Adding JSONSchema Validator

### CODE REVIEW

File: SiftSdk.podspec [hunk 1]
```diff
@@ -15,10 +15,12 @@ Pod::Spec.new do |spec|
   spec.platform     = :ios, "12.0"
   spec.swift_version = "5.2.4"
 
-  spec.source       = { :git => "https://github.com/thedatasherpas/SIFT_SDK_iOS.git", :tag => "1.0.2" }
+  spec.source       = { :git => "https://github.com/thedatasherpas/SIFT_SDK_iOS.git", :tag => "1.1.0" }
 
   spec.source_files  = "SiftSdk", "SiftSdk/**/*.{h,m,swit}"
```
Comment (LamarDataSherpas) [inline SiftSdk.podspec:20]: What is swit?

File: SiftSdk.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -90,6 +91,7 @@
 				2A94A6F9252A001500AF04BB /* SiftSdkTests */,
 				2A7F800F25490AA4001A1983 /* SiftSdkTests */,
 				2A94A6ED252A001500AF04BB /* Products */,
+				909D32D56757DF94F3D22AA0 /* Pods */,
```
Comment (LamarDataSherpas) [inline SiftSdk.xcodeproj/project.pbxproj:94]: remove all Pods stuff from project file

File: SiftSdk/SiftSdkExtension.swift [hunk 3]
```diff
@@ -66,7 +67,12 @@ extension SiftSdk: SPSiftCallback {
     if let datas = dictionary[kSPData] as? [[String: Any]] {
       for data in datas {
         let dictToSend = getDictionaryToSend(dictionary: data, eventSchema: &eventSchema)
-        array.append(dictToSend)
+        let jsonSchemaValidator = JSONSchema.validate([dictToSend], schema: defaultSchema())
+        if jsonSchemaValidator.valid {
+            array.append(dictToSend)
+        } else {
+            NSLog("ERROR! invalid schema \(jsonSchemaValidator.errors?.description ?? "") \(#function):\(#line)")
```
Comment (LamarDataSherpas) [inline SiftSdk/SiftSdkExtension.swift:74]: bug. What if there are 2 events and 1 of them does not pass validation. The client will only get 1 call back.

Comment (QasimAbbas) [reply SiftSdk/SiftSdkExtension.swift:74]: This validates per dict.  This should still give the error here regardless before it's sent to Snowplow so is that necessary?

Comment (QasimAbbas) [reply SiftSdk/SiftSdkExtension.swift:74]: Updated code to accommodate callback

**Review by LamarDataSherpas — APPROVED** [review]

File: SiftSdk.xcodeproj/project.pbxproj (hunk 2 above)
Comment (QasimAbbas) [reply SiftSdk.xcodeproj/project.pbxproj]: This has been removed

**Pull request merged**